server/jamovi/server/dataset/duckdataset.py

```diff
@@ -172,32 +216,206 @@ def __iter__(self) -> Iterator[DuckColumn]:
         for column in self._columns_by_index:
             yield column
 
-    def set_value(self, row: int, column: int, value, initing=False):
+    def set_value(
+        self,
+        row: int,
+        column: int | DuckColumn,
+        raw_value: int | float | str,
+        initing=False,
+    ):
         """set a value in the data set"""
-        col = self._columns_by_index[column]
 
-        if isinstance(value, str) and col.data_type is DataType.TEXT and col.has_levels:
-            try:
-                raw = col.get_value_for_label(value)
-            except KeyError:
-                raw = self.column_add_level(col, value)
-            value = raw
+        if isinstance(column, int):
+            column = self._columns_by_index[column]
 
+        if column.data_type is DataType.DECIMAL:
+            assert isinstance(raw_value, Number)
+        elif (
+            column.data_type is DataType.TEXT and column.measure_type is MeasureType.ID
+        ):
+            assert isinstance(raw_value, str)
+        else:
+            assert isinstance(raw_value, int)
```

Comment:
What's the use of these assertions here?

Reply:
these are to catch subtle bugs ... an issue with SQL is that it will fail silently, in a way that the shared memory wouldn't ...